pkg/reconcilermanager/controllers/jsonpath.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/spyzhov/ajson"
+	"gopkg.in/yaml.v3"
+	"k8s.io/klog/v2"
+)
+
+// deleteFields evaluates the JSONPath expression to delete fields from the input map.
+func deleteFields(obj map[string]interface{}, expression string) error {
```

Comment:
Can you make a follow up to add a Delete function to the cli-utils jsonpath, which already has Get and Set funcs?

https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/jsonpath/jsonpath.go

It would be nice to keep them all together and just use that package here.

Reply:
Yes, sounds good.